daos-stack/daos
# DAOS-8824 test: Automate dmg telemetry test for dkey/akey enum and punch

Added a new test file, telemetry/dkey_akey_enum_punch.py

test_dkey_akey_enum_punch()
Test count and active for enum and punch.

Test Steps:
1. Write 100 objects.
2. Insert 2 dkeys per object. Insert 1 akey per dkey. Use OC_S1.
3. Punch all akeys and dkeys.
4. Call list_dkey() and list_akey() on the objects.
5. Verify the metrics.

test_tgt_dkey_akey_punch()
Test punch count for tgt values.

Test Steps:
1. Write 100 objects.
2. Insert 2 dkeys per object. Insert 1 akey per dkey. Use OC_RP_2G1.
3. Punch all akeys and dkeys.
4. Verify the metrics.

Also refactored telemetry_test_base.py

Skip-unit-tests: true
Test-tag: dkey_akey_enum_punch tgt_dkey_akey_punch
Signed-off-by: Makito Kano <makito.kano@intel.com>
